### DISCUSSION TITLE
Question: How to click on all detected ads

### DISCUSSION THREAD

**Discussion by tezzat10** [dhowe/AdNauseam#1905] [General]
Hello 

I want the extension to click on all ads how to do that, please?

Also is it possible to click on all ads and make the ads open in new tabs I know it will be annoying having all of the tabs opened everytime I visit a website with ads on it but I wanna make sure and browse these ads 

My case is like spying on my competitors kinda solution 

Thank you so much for this amazing extension 

waiting for your reply 

Thanks in advance 

Stay safe 

Cheers

**Comment by tezzat10** [comment]
Could you please help me ASAP

the extension isn't working at all right now

**Comment by dhowe** [comment]
You can click on all ads, by setting the click frequency to 'always' as shown below. The only way to see each ad would be to go into either the the ad-vault or the menu display and click on each.

![image](https://user-images.githubusercontent.com/737638/127703508-27f1dafb-4ecb-47d0-aad6-ef0e8eb70e39.png)

**Reply by tezzat10** [reply]
Hello 

Thank you so much for your help 

It's already on Always that's why I'm not sure why it's not working at all 

I even tried to remove all of the other extensions on my chrome cause I thought maybe one of the extensions is conflicting with AdNauseam so I tried everything 

It doesn't click on Google Adsense or any other ads from any other provider at all don't know why 

the ad vault counter doesn't increase at all it's always on ZERO 

So I don't really know what is wrong with the extension and why it's not working at all 

so what do you think might be the problem? 

Waiting for your reply 

Thanks in advance 

Cheers

**Comment by dhowe** [comment]
Please tell us your OS, browser version, and version of AdNauseam. Thanks

**Reply by tezzat10** [reply]
Sure 

My OS is Windows 10 and my browser version is : Version 92.0.4515.107 (Official Build) (64-bit)

and AdNauseam Version is  adnauseam-3.11.4.chromium.zip

Waiting for your reply 

Cheers

**Reply by tezzat10** [reply]
Opera doesn't work either 

I just installed it on Opera and nothing happens 

Waiting for your reply 

Cheers

**Comment by dhowe** [comment]
Can you visit [this page](http://lab-lamp.scm.cityu.edu.hk/adntest/simple.html) with both versions and let us know what you see?

**Reply by tezzat10** [reply]
it's working just fine on that page 

but for example these two websites 

Myip.com has Google Adsense ads

Or businessxm.com has a different provider 

It doesn't click on anything on it and the counter is stuck on ZERO

so what do you think ?

waiting for your reply

Cheers

**Reply by mneunomne** [reply]
Hello @tezzat10,

Thanks for the feedback!

Yes indeed it is not all ads that we are able to click unfortunatelly. We can only click on the ones that are hidden by cosmetic filters, and it is not always that we can manage to have them block onlyed with cosmetic rules. Often we just have to block the requests alltogether. And there are other cases where clicking is not possible at the moment/

Regarding these two examples that you provided, indeed it seems that it is not able to collect or click them. We'll work to try to improove ad collection and clicking in both of the ad providers that you shared. 

Just so you understand the scope of our work, the ads providers are often changing their methods and code structures in showing us ads. And for us to collect and click them, we need to be constantly be updating to this forever chaging landscape. 
But if you are interested, you can also create your own cosmetic rules for ads, if they are made correctly, you will be able to collect and click them.

If you are new to Adn or Ublock you can check [here](https://github.com/gorhill/uBlock/wiki/Introduction-to-basic-filtering-syntax) more about how cosmetic filter syntax works. And [here](https://github.com/dhowe/AdNauseam/wiki/Developer-FAQ#how-does-adnauseam-detect-ads) how our ad collection work. 

Hope this somewhat helps you!

Best,
Alberto

**Comment by dhowe** [comment]
[DELETED MESSAGE]
@tezzat10, please use email or other means for personal messages, thanks

**Reply by tezzat10** [reply]
so sorry it's my first time here on github I still don't know how to send a Personal message 

Anyway I just sent you an email Alberto on your email listed at your profile 

Waiting for your reply 

Cheers
